src/main/java/sx/blah/discord/handle/impl/obj/User.java

```diff
@@ -211,7 +212,7 @@ public void setDiscriminator(String discriminator) {
 		if (roles != null) {
 			RolesHolder retrievedRoles = roles.get(guild.getLongID());
 			if (retrievedRoles != null && retrievedRoles.getObject() != null)
-				return new LinkedList<>(retrievedRoles.getObject());
+				return new LinkedList<>(retrievedRoles.getObject().stream().filter(Objects::nonNull).collect(Collectors.toList()));
```

Comment:
I think there is a bigger underlying issue if for some reason we are storing null roles in the first place. While I'm ok with bandaid fixes because we'd rather just focus on the development of v3, other devs should add input.

Additionally, this should be changed so you do a stream then collect to list, rather than collect to list while putting it in another list.